crates/environ/src/vmoffsets.rs

```diff
@@ -181,103 +182,63 @@ impl<P: PtrSize> From<VMOffsetsFields<P>> for VMOffsets<P> {
-            .store
-            .checked_add(u32::from(ret.ptr.size() * 2))
-            .unwrap();
-        ret.imported_functions = ret
-            .signature_ids
-            .checked_add(u32::from(ret.ptr.size()))
-            .unwrap();
-        ret.imported_tables = ret
-            .imported_functions
-            .checked_add(
-                ret.num_imported_functions
-                    .checked_mul(u32::from(ret.size_of_vmfunction_import()))
-                    .unwrap(),
-            )
-            .unwrap();
-        ret.imported_memories = ret
-            .imported_tables
-            .checked_add(
-                ret.num_imported_tables
-                    .checked_mul(u32::from(ret.size_of_vmtable_import()))
-                    .unwrap(),
-            )
-            .unwrap();
-        ret.imported_globals = ret
-            .imported_memories
-            .checked_add(
-                ret.num_imported_memories
-                    .checked_mul(u32::from(ret.size_of_vmmemory_import()))
-                    .unwrap(),
-            )
-            .unwrap();
-        ret.defined_tables = ret
-            .imported_globals
-            .checked_add(
-                ret.num_imported_globals
-                    .checked_mul(u32::from(ret.size_of_vmglobal_import()))
-                    .unwrap(),
-            )
-            .unwrap();
-        ret.defined_memories = ret
-            .defined_tables
-            .checked_add(
-                ret.num_defined_tables
-                    .checked_mul(u32::from(ret.size_of_vmtable_definition()))
-                    .unwrap(),
-            )
-            .unwrap();
-        ret.defined_globals = align(
-            ret.defined_memories
-                .checked_add(
-                    ret.num_defined_memories
-                        .checked_mul(u32::from(ret.size_of_vmmemory_definition()))
-                        .unwrap(),
-                )
-                .unwrap(),
-            16,
-        );
-        ret.defined_anyfuncs = ret
-            .defined_globals
-            .checked_add(
-                ret.num_defined_globals
-                    .checked_mul(u32::from(ret.size_of_vmglobal_definition()))
-                    .unwrap(),
-            )
-            .unwrap();
-        ret.builtin_functions = ret
-            .defined_anyfuncs
-            .checked_add(
-                ret.num_imported_functions
-                    .checked_add(ret.num_defined_functions)
-                    .unwrap()
-                    .checked_mul(u32::from(ret.size_of_vmcaller_checked_anyfunc()))
-                    .unwrap(),
-            )
-            .unwrap();
-        ret.size = ret
-            .builtin_functions
-            .checked_add(u32::from(ret.pointer_size()))
-            .unwrap();
+        // Use a local variable instead of ret.size to prevent unnecessary loads and stores.
+        // LLVM is not able to optimize accesses of ret.size away most of the time.
```

Comment:
This comment I think is fine to drop because it basically references something not existent in the code after this PR.